src/main/java/jenkinsci/plugins/influxdb/models/Target.java

```diff
@@ -181,5 +199,59 @@ public FormValidation doCheckUrl(@QueryParameter String value) {
         public FormValidation doCheckDatabase(@QueryParameter String value) {
             return FormValidation.validateRequired(value);
         }
+        
+        @POST
+        public FormValidation doVerifyConnection(@QueryParameter String url, @QueryParameter String credentialsId,
+                                                 @QueryParameter String organization, @QueryParameter String database,
+                                                 @QueryParameter String retentionPolicy, @AncestorInPath Item context) {
+            InfluxDBClient influxDB = null;
+            doCheckUrl(url);
+            doCheckDatabase(database);
+            try {
+                Jenkins.get().checkPermission(Jenkins.ADMINISTER);
+
+                List<StandardUsernamePasswordCredentials> lookupCredentials = CredentialsProvider.lookupCredentials(
+                        StandardUsernamePasswordCredentials.class,
+                        context,
+                        ACL.SYSTEM,
+                        URIRequirementBuilder.fromUri(url).build());
+
+                StandardUsernamePasswordCredentials credentials = CredentialsMatchers.firstOrNull(lookupCredentials, CredentialsMatchers.withId(null == credentialsId ? "" : credentialsId));
+
+                if (organization != null && !organization.trim().isEmpty()) {
+                    InfluxDBClientOptions.Builder options = InfluxDBClientOptions.builder().url(url).org(organization).bucket(database);
+
+                    if (credentials != null) { // basic auth
+                        options.authenticate(credentials.getUsername(), credentials.getPassword().getPlainText().toCharArray());
+                    } else { // token auth
+                        List<StringCredentials> lookupTokenCredentials = CredentialsProvider.lookupCredentials(//
+                                StringCredentials.class,//
+                                context,//
+                                ACL.SYSTEM,//
+                                URIRequirementBuilder.fromUri(url).build());//
+                        StringCredentials c = CredentialsMatchers.firstOrNull(lookupTokenCredentials, CredentialsMatchers.withId(null == credentialsId ? "" : credentialsId));
+                        assert c != null;
+                        options.authenticateToken(c.getSecret().getPlainText().toCharArray());
+                    }
+
+                    influxDB = InfluxDBClientFactory.create(options.build());
+
+                } else {
+                    influxDB = credentials == null
+                            ? InfluxDBClientFactory.createV1(url, "", "".toCharArray(), database, retentionPolicy)
+                            : InfluxDBClientFactory.createV1(url, credentials.getUsername(),
+                                    credentials.getPassword().getPlainText().toCharArray(), database, retentionPolicy);
+                }
+
+                Bucket bucket = influxDB.getBucketsApi().findBucketByName(database);
```

Comment:
Does this work with InfluxDB 1.8 targets? I get a long error stack trace every time I try to test the connection of a 1.8 target.

Reply:
Also 2.x validation fails with
```
java.lang.NoSuchMethodError: 'com.google.gson.stream.JsonReader com.google.gson.Gson.newJsonReader(java.io.Reader)'
	at retrofit2.converter.gson.GsonResponseBodyConverter.convert(GsonResponseBodyConverter.java:38)
	at retrofit2.converter.gson.GsonResponseBodyConverter.convert(GsonResponseBodyConverter.java:27)
	at retrofit2.OkHttpCall.parseResponse(OkHttpCall.java:243)
	at retrofit2.OkHttpCall.execute(OkHttpCall.java:204)
	at com.influxdb.internal.AbstractRestClient.execute(AbstractRestClient.java:78)
	at com.influxdb.client.internal.BucketsApiImpl.findBucketByName(BucketsApiImpl.java:93)
	at jenkinsci.plugins.influxdb.models.Target$DescriptorImpl.doVerifyConnection(Target.java:246)
	...```

Reply:
Sorry, I don't have many influxDB versions... I test only on my own (I work with influxDB 2.2.0).

Reply:
Yes, but as you can see `findBucketByName` uses `retrofit2.OkHttpCall` instead of `okhttp3.OkHttpClient` (as used by the plugin elsewhere). The issue is with that one particular call `Bucket bucket = influxDB.getBucketsApi().findBucketByName(database);` which is causing this issue and as the plugin also needs to support InfluxDB 1.8, it's not succeeding with that.

Reply:
I did an update to only call a "ping", but I don't understand the error you get because elsewhere in the plugin I found use of retrofit2 :

```
    @Nonnull
    @Override
    public WriteApiBlocking getWriteApiBlocking() {
        return new WriteApiBlockingImpl(retrofit.create(WriteService.class), options);
    }
```

This is the api used to write point...

Reply:
Thank you, now both InfluxDB 1.8 and 2.2 targets work without any error traces in logs either.